pkg/tasks/security/authorization/externalauthorization.go

```diff
@@ -0,0 +1,149 @@
+	util.Log.Info("Verify that sleep can access httpbin")
+	if !strings.Contains(msg, "200") {
+		util.Log.Errorf("Verify setup -- Unexpected response code: %s", msg)
+	} else {
+		util.Log.Infof("Success. Get expected response: %s", msg)
+	}
+
+	t.Run("Deploy the External Authorizer", func(t *testing.T) {
+
+		util.Log.Info("Deploy the sample External Authorizer")
+		util.KubeApplyContents("foo", ExternalAuthzService)
+		time.Sleep(time.Duration(50) * time.Second)
+
+		util.Log.Info("Verfiy the sample external authorizer is up and running")
+		extAuthPod, err := util.GetPodName("foo", "app=ext-authz")
+		util.Inspect(err, "Failed to get ext-authz pod name", "", t)
+		extAuthPodverf, err := util.Shell("oc logs %s -n foo -c ext-authz", extAuthPod)
+
+		if strings.Contains(extAuthPodverf, "Starting HTTP server at [::]:8000") && strings.Contains(extAuthPodverf, "Starting gRPC server at [::]:9000") {
+			util.Log.Infof("Success the sample external authorizer is up and running")
+		} else {
+			util.Log.Errorf("Failed the sample external authorizer is not up and not running")
+		}
+
+	})
+	t.Run("Define the external authorizer", func(t *testing.T) {
+
+		util.Log.Info("Edit configmap for two external providers")
+		util.Shell(`kubectl patch -n %s configmap/istio-%s --type merge -p '{"data": {"mesh": "# Add the following content to define the external authorizers.\nextensionProviders:\n- name: \"sample-ext-authz-grpc\"\n  envoyExtAuthzGrpc:\n    service: \"ext-authz.foo.svc.cluster.local\"\n    port: \"9000\"\n- name: \"sample-ext-authz-http\"\n  envoyExtAuthzHttp:\n    service: \"ext-authz.foo.svc.cluster.local\"\n    port: \"8000\"\n    includeRequestHeadersInCheck: [\"x-ext-authz\"]"}}'`, meshNamespace, smcpName)
+	})
+
+	t.Run("Enable with external authorization", func(t *testing.T) {
+
+		util.Log.Info("Enable with external authorization")
+		util.KubeApplyContents("foo", ExternalRoute)
+
+		util.Log.Info("Verfiy the header with deny server")
+		deny := fmt.Sprintf(`curl "http://httpbin.foo:8000/headers" -H "x-ext-authz: deny" -s`)
+		extAuthDeny, err := util.Shell("kubectl exec %s -c sleep -n foo -- %s", sleepPod, deny)
+		util.Inspect(err, "Failed to run the command", "", t)
+		if strings.Contains(extAuthDeny, "denied by ext_authz for not found header `x-ext-authz: allow` in the request") {
+			util.Log.Infof("Success, verfication the header with deny server")
+		} else {
+			util.Log.Errorf("verfication failed with deny server")
+		}
+
+		util.Log.Info("Verfiy the header with allow server")
+
+		type Headers struct {
+			Accept     string
+			Host       string
+			UserAgent  string `json:"User-Agent"`
+			XB3Sampled string `json:"X-B3-Sampled"`
+			XB3Spanid  string `json:"X-B3-Spanid"`
+			XB3Traceid string `json:"X-B3-Traceid"`
+			XExtAuthz  string `json:"X-Ext-Authz"`
+		}
+
+		type Response struct {
+			Headers Headers
+		}
+
+		var response Response
+		util.Inspect(err, "Failed to get sleep pod name", "", t)
```

Comment:
`err` was last set by line 96 and does not indicate an error getting the sleep pod name (likely just a copy-paste mistake from lines 53-54). The same issue occurs on line 131.